File: android-app/feature/feature-discover/src/main/java/tv/bayit/plus/feature/discover/model/FeatureAvailabilityState.kt
package tv.bayit.plus.feature.discover.model

sealed class FeatureAvailabilityState {
    data object Ready : FeatureAvailabilityState()

    data class SetupNeeded(
        val missing: List<FeaturePrerequisite>,
    ) : FeatureAvailabilityState()

    data object PremiumRequired : FeatureAvailabilityState()

    data class NotAvailable(
        val reasonKey: String,
    ) : FeatureAvailabilityState()

    data class PlatformOnly(
        val platform: DiscoverPlatform,
    ) : FeatureAvailabilityState()

    val badgeColorName: String
        get() = when (this) {
            is Ready -> "green"
            is SetupNeeded -> "orange"
            is PremiumRequired -> "purple"
            is NotAvailable -> "gray"
            is PlatformOnly -> "blue"
        }

    val badgeLabelKey: String
        get() = when (this) {
            is Ready -> "discover.badge.ready"
            is SetupNeeded -> "discover.badge.setupNeeded"
            is PremiumRequired -> "discover.badge.premium"
            is NotAvailable -> "discover.badge.notAvailable"
            is PlatformOnly -> "discover.badge.platformOnly"
        }
}

File: ios-app/Packages/BayitCore/Sources/BayitCore/Discover/FeatureAvailabilityState.swift
import Foundation

public enum FeatureAvailabilityState: Sendable {
    case ready
    case setupNeeded([FeaturePrerequisite])
    case premiumRequired
    case notAvailable(reasonKey: String)
    case platformOnly(Platform)

    public var badgeColorName: String {
        switch self {
        case .ready: return "green"
        case .setupNeeded: return "orange"
        case .premiumRequired: return "purple"
        case .notAvailable: return "gray"
        case .platformOnly: return "blue"
        }
    }

    public var badgeLabelKey: String {
        switch self {
        case .ready: return "discover.badge.ready"
        case .setupNeeded: return "discover.badge.setupNeeded"
        case .premiumRequired: return "discover.badge.premium"
        case .notAvailable: return "discover.badge.notAvailable"
        case .platformOnly: return "discover.badge.platformOnly"
        }
    }
}
